src/libs/outsideClick/index.js

```diff
@@ -0,0 +1,33 @@
+class OutsideClick {
+    constructor() {
+        this.containerRef = null;
+        this.callback = null;
+        this.listening = false;
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    handleClick(event) {
+        if (!this.containerRef || !this.callback) return;
```

Comment:
if in startListening we made this check then we don't need it again, also if this mehtod is private let's use:
```
const privateHandleClick = Symbol('hanldeClick');

class OutsideClick {
    ...
    [privateHandleClick]() {}
    ...
    document.addEventListener('click', this[privateHandleClick]);
}
```
in order to hide it from outside